numpy/compat/py3k.py

```diff
@@ -18,76 +18,49 @@
 
 import sys
 import os
-try:
-    from pathlib import Path, PurePath
-except ImportError:
-    Path = PurePath = None
-
-if sys.version_info[0] >= 3:
-    import io
-
-    try:
-        import pickle5 as pickle
-    except ImportError:
-        import pickle
+from pathlib import Path, PurePath
+from importlib.machinery import SourceFileLoader
```

Comment:
We should leave this import in `npy_load_module`, to save paying for an import of `importlib` at startup.

Reply:
should this be public??? I guess we are stuck with it. It seems to only be used in setup.py

Reply:
We should probably replace its callers with direct usages of `importlib` within numpy, then deprecate it - but not as part of this patch.